spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/support/DefaultSubscriberFactory.java

```diff
@@ -282,4 +321,85 @@ public SubscriberStub createSubscriberStub() {
 		}
 	}
 
+	/**
+	 * Creates {@link ExecutorProvider}. If a custom executor provider is set then the
+	 * subscriber properties configured through the application.properties file will be
```

Comment:
Why did you decide to have the custom `ExecutorProvider` to take precedence over subscription-specific config? I would expect it to be the opposite.

Reply:
Yes, that is a valid point! Previously, the auto-config subscriber properties would be taken into account only when the custom bean was missing (as indicated by the `ConditionalOnMissingBean` annotation on our bean). This new logic follows that previous situation where the custom bean takes first priority over the auto-config (global or subscription-specific) properties. 

We also considered the reverse scenario where the subscription-specific would take precedence over the custom bean but the implementation details turned out to be tricky. We would have to add extra logic in `PubSubConfiguration` to find some way to determine if the property was set and different from the default as well as the global setting. This would have to be done for not just one property but all the other properties to come. As a compromise, the resolution was to allow custom beans to take precedence and if one would like to use subscription-specific properties then they will need to have to avoid using the custom bean. 

This change also needs to be carefully documented.

Reply:
Should we open an issue to track documentation/clarification of this?

Reply:
Good point, done.